lib/internal/Magento/Framework/Shell/Test/Unit/CommandRendererTest.php

```diff
@@ -9,45 +9,37 @@
 
 class CommandRendererTest extends \PHPUnit\Framework\TestCase
 {
-    public function testRender()
+    /**
+     * @param $expectedCommand
+     * @param $actualCommand
+     * @param $testArguments
+     * @dataProvider testRenderDataProvider
+     */
+    public function testRender($expectedCommand, $actualCommand, $testArguments)
     {
-        $testArgument  = 'argument';
-        $testArgument2 = 'argument2';
         $commandRenderer = new CommandRenderer();
-        $expectedCommand = "php -r " . escapeshellarg($testArgument) . " 2>&1 | grep " . escapeshellarg($testArgument2) . " 2>&1";
-        $this->assertEquals(
-            $expectedCommand,
-            $commandRenderer->render('php -r %s | grep %s', [$testArgument, $testArgument2])
-        );
-
         $this->assertEquals(
             $expectedCommand,
-            $commandRenderer->render('php -r %s 2>&1 | grep %s', [$testArgument, $testArgument2])
-        );
-
-        $this->assertEquals(
-            $expectedCommand,
-            $commandRenderer->render('php -r %s 2>&1 2>&1 | grep %s', [$testArgument, $testArgument2])
+            $commandRenderer->render($actualCommand, $testArguments)
         );
     }
 
-    public function testRenderWithoutArguments()
+    public function testRenderDataProvider()
```

Comment:
Don't start data provider name with `test`. Describe what it actually returns, like `commandsDataProvider`.

Reply:
My fault, I'm changing that